napari/_qt/qt_main_window.py

```diff
@@ -385,6 +386,10 @@ def __init__(self, viewer, *, show: bool = True):
         self._add_plugins_menu()
         self._add_help_menu()
 
+        # discover any themes provided by plugins
+        plugin_manager.discover_themes()
```

Comment:
other than triggering discovery when we show the first window, it's not immediately clear to me why all of this logic now lives on the mainwindow.    I feel like we're mixing business logic from the plugin_manager and the settings altogether on the main window here.  most of the new theme-related methods aren't really using the `self` attribute (except to call `_rebuild_theme_settings`, which is just a static method that perhaps belongs on the settings manager instead?)

Reply:
I've removed a couple of the methods since they were not really required. 

The remaining methods such as `_setup_existing_themes`, `_add_theme`, `_remove_theme`, `_connect_theme` and `_theme_icon_changed` methods are added in the `MainWindow` so any changes to the theme values are reflected in the GUI.